packages/client/src/pages/CreateSafe.js

```diff
@@ -107,9 +107,16 @@ function CreateSafe({ web3 }) {
     submittedTransaction,
     createTreasury,
   } = web3;
-  const [safeOwners, setSafeOwners] = useState([
-    { name: "", address, verified: true },
-  ]);
+  const [safeOwners, setSafeOwners] = useState();
+
+  useEffect(() => {
+    if (address) {
+      setSafeOwners([
+        { name: "", address, verified: true },
+      ])
+    }
+  }, [address]);
```

Comment:
If for some reason address changes, we will reset all the safeOwner data to `[{ name: "", address, verified: true }]` is this the expected behavior?

Reply:
Address wont change, this is just a prevention for the one point in time where address is not yet loaded and is undefined

Reply:
Got it, in this case we should prevent this page from rendering unless we have address ready. Using useEffect will create unexpected side effects, like you mentioned, it could change from undefined to defined, what about the situation of it changing from defined to undefined?

Reply:
If it's to deal with the catching up situation that only happens on the initial rendering, it's cheaper and more reliable to do optional rendering of the CreateSafe component since address is required to populate the page.

Reply:
I just pushed a different solution, with conditional rendering
In order to accomplish that I had to add a helper prop to the user 'loaded'
This helper is needed beside the address check, because when the user is not logged in - the address is undefined, then the home page won't load
The other attribute that I considered is 'loggedIn', but this one is too undefined due to async fetching
Let me know your thoughts :)

Reply:
I prefer the idea of using the `loggedIn`, since address is critical in creating a safe and loading a safe, preventing the home page from loading when address is absent is the exact behavior we should expect in my opinion.
Also I would suggest move this check logic from Web3 into Home.js line 36. Web3 is an HOC, having the check logic there will impact all the child components using it, some of them don't care about if user is logged in or has an address.

Reply:
It's not related to the Home component, it's related to the CreateSafe, so it won't work if we move the logic there.
It's not affecting the logic at all, it is just stopping it from re-rendering more times than needed - so we need to make sure the data is loaded before we render anything -> so either the user is loaded (when the user is not logged in) or the address is loaded (when the user is logged in)